mmtc-core/src/main/java/edu/jhuapl/sd/sig/mmtc/cfg/MmtcConfig.java

```
@@ -896,6 +896,27 @@ public Integer getSclkScetApplicableDurationDays() {
         return timeCorrelationConfig.getConfig().getInt("product.sclkScetFile.applicableDurationDays", 0);
     }
 
+    /**
+     * Gets the number of decimal places/fractional digits to display when recording oscillator temperature in the
+     * time history file
+     * @return the integer specified in the config unless key isn't present or can't be parsed, then return safe default
+     */
+    public int getOscTempFractionDigits() {
+        final String keyName = "table.timeHistoryFile.oscTempFractionDigits";
+        int numDigits = 16; // selected as the (approximately) largest number of fractional digits in a normal Java double
+                            // before floating point precision errors take over
+        try {
+            if (timeCorrelationConfig.getConfig().containsKey(keyName)) {
+                numDigits = timeCorrelationConfig.getConfig().getInt(keyName);
+            } else {
+                logger.debug("Key {} not found in TimeCorrelationConfigProperties: defaulting to value {}", keyName, numDigits);
+            }
+        } catch (Exception e) {
+            logger.warn("Failed to parse integer provided in key {}, defaulting to {}", keyName, numDigits);
```

Let's have this throw instead of just log, so the user can realize more quickly they've made a mistake in their intended configuration.

---

I appreciate your line of thinking!  I think (hope) having a sane default value will help folks get to a successful MMTC run quickly (by just omitting the key entirely).  In my experience so far, failing early will shorten the overall time it takes to get to the desired configuration, instead of delaying it til later (e.g. when a user actually takes time to read the stdout, or is actually looking at a Time History File.)